src/include/migraphx/operators.hpp

```diff
@@ -581,23 +580,28 @@ struct reshape
         {
             if(dims[i] == 0)
                 rdims[i] = idims[i];
+
+            // since rdims using size_t type, -1 is the max value
+            // is size_t that cause later compuation incorrect
```

Comment:
Why dont we change rdims to `int64_t` instead? Then we can just write:

```
auto rdims = dims;
```

Reply:
rdims is used to initialize the return shape that needs rdims to be size_t (int64_t will cause build error)